src/Mvc/Mvc.Testing/src/WebApplicationFactory.cs

```diff
@@ -360,20 +392,13 @@ private void EnsureDepsFile()
         protected virtual IWebHostBuilder CreateWebHostBuilder()
         {
             var builder = WebHostBuilderFactory.CreateFromTypesAssemblyEntryPoint<TEntryPoint>(Array.Empty<string>());
-            if (builder == null)
-            {
-                throw new InvalidOperationException(Resources.FormatMissingBuilderMethod(
-                    nameof(IHostBuilder),
-                    nameof(IWebHostBuilder),
-                    typeof(TEntryPoint).Assembly.EntryPoint!.DeclaringType!.FullName,
-                    typeof(WebApplicationFactory<TEntryPoint>).Name,
-                    nameof(CreateHostBuilder),
-                    nameof(CreateWebHostBuilder)));
-            }
-            else
+
+            if (builder is null)
             {
-                return builder.UseEnvironment(Environments.Development);
+                return new NullWebHostBuilder();
```

Comment:
I could also change the nullability here but I couldn't figure out how to do that. All of the tooling was fighting me.

Reply:
So changing the method signature to:

```
protected virtual IWebHostBuilder? CreateWebHostBuilder()
```

Is not working here? Bummer.

I say that mostly because it tripped me up while reading it. Why are we creating a `NullWebHostBuilder`? Well, that doesn't actually matter, we're just saying there's no builder on the target assembly so we need to construct our own deferred one. It would've read better with nullability set correctly here. If it's possible to do that, I think it'll help us with maintainability for this moving forward.

Reply:
I would actually rather change it to be that but I couldn't get the public API files working. I would need somebody to explain the error I was getting 😄 . Also I don't know what the bar is for breaking nullability changes.

cc @pranavkm @JamesNK

Reply:
Yes, I was raging at this last night and I went to bed after outsmarting nullable 😄